src/dmd/dsymbolsem.d

```diff
@@ -3628,6 +3761,55 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
         funcDeclarationSemantic(funcdecl);
     }
 
+    override void visit(CopyCtorDeclaration cctd)
+    {
+        //printf("CopyCtorDeclaration::semantic() %s\n", toChars());
+        if (cctd.semanticRun >= PASS.semanticdone)
+            return;
+
+        if (cctd._scope)
+        {
+            sc = cctd._scope;
+            cctd._scope = null;
+        }
+
+        cctd.parent = sc.parent;
+        Dsymbol p = cctd.toParent2();
+        StructDeclaration sd = p.isStructDeclaration();
+        if (!sd)
+        {
+            error(cctd.loc, "copy constructor can only be a member of aggregate, not %s `%s`", p.kind(), p.toChars());
```

Comment:
Aggregates include classes, which don't have copy constructors. Error message is erroneous.